File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_75775.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request
import json


async def BenchmarkTest75775(request: Request):
    graphql_var = json.loads((await request.body()).decode()).get('variables', {}).get('input', '')
    request.session['data'] = str(graphql_var)
    return {"updated": True}

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest75775.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.security.*;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest75775 {

    @PostMapping(path="/BenchmarkTest75775", consumes="application/xml")
    public void BenchmarkTest75775(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        StringBuilder carrier = new StringBuilder();
        carrier.append(xmlValue);
        String data = carrier.toString();
        byte[] digest = MessageDigest.getInstance("MD5").digest(data.getBytes());
        response.setHeader("X-Hash", java.util.Base64.getEncoder().encodeToString(digest));
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
